cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -590,18 +656,21 @@ private SearchResults searchByDefaultIndex(String namespaceId, String searchQuer
     return new SearchResults(results, Collections.emptyList());
   }
 
-  private SearchResults searchByCustomIndex(String namespaceId, Set<EntityTypeSimpleName> types,
-                                            SortInfo sortInfo, int offset, int limit, int numCursors,
-                                            @Nullable String cursor, boolean showHidden,
-                                            Set<EntityScope> entityScope) {
+  private SearchResults searchByCustomIndex(SearchRequest request) {
+    SortInfo sortInfo = request.getSortInfo();
+    String cursor = request.getCursor();
+    int offset = request.getOffset();
+    int limit = request.getLimit();
+    int numCursors = request.getNumCursors();
+
     List<MetadataEntry> results = new LinkedList<>();
     String indexColumn = getIndexColumn(sortInfo.getSortBy(), sortInfo.getSortOrder());
     // we want to return the first chunk of 'limit' elements after offset
     // in addition, we want to pre-fetch 'numCursors' chunks of size 'limit'.
     // Note that there's a potential for overflow so we account by limiting it to Integer.MAX_VALUE
     int fetchSize = (int) Math.min(offset + ((numCursors + 1) * (long) limit), Integer.MAX_VALUE);
     List<String> cursors = new ArrayList<>(numCursors);
-    for (String searchTerm : getSearchTerms(namespaceId, "*", entityScope)) {
+    for (String searchTerm : getSearchTerms(request.getNamespace().getNamespace(), "*", request.getEntityScope())) {
```

Comment:
Should we move the getSearchTerm logic to SearchRequest. 

Maybe later on we can even simplify this by having getSearchTerm as a interface method where different SearchRequest like PreferixedSearchRequest can create their search terms and encapsulate to logic rather than floating around as conditional statement everywhere.

Reply:
perhaps, but I would rather do that later once I get a better handle on what that method is doing. I get the feeling it is pretty specific to how the dataset is storing its data, in which case it belongs here and not in the SearchRequest.